### ISSUE TITLE
Dealing with Duplicate Authors

### ISSUE THREAD

**Issue by formerlytomato** [bookwyrm-social/bookwyrm#1119]
Currently an author can be registered separately as [Soren Kierkegaard](https://bookwyrm.social/author/6166), [Søren Kierkegaard](https://bookwyrm.social/author/18619), and  [Sören Kierkegaard](https://bookwyrm.social/author/8779), all of which are treated separately and will have books split among them.

Giving users the ability to mark Soren Kierkegaard and Sören Kierkegaard as "aliases" of Søren Kierkegaard would make the cataloguing a lot cleaner.

**Comment by kopischke** [comment]
With apologies for chiming into the issue, I’d like to make the observation that BookWyrm is already halfway there: it is possible to add aliases to an author – they just aren’t hooked up to anything. What is missing is an author representation that integrates aliases, so that any version of the name points to the same author page. I have no idea by how much that would complicate the model, so I will refrain from judging on how hard this would be to implement, but I will add that there already is a workflow for confirming this linking with the “is this a known author” dialog BookWyrm presents when editing a book’s author, so the change in the model is essentially all that is needed AFAICS.

**Comment by hughrun** [comment]
#1581 should reduce the frequency of this problem but doesn't actually provide a way to merge already-existing records. I'll have more of a think about that: it's probably something we'd want admins to do rather than just anyone, but it needs to be easym (like a checkbox yes/no), and probably should use some kind of scheduled or run-on-demand background task.

**Comment by kopischke** [comment]
Just spitballing here, but one thing we might be able to leverage is the existing alias system, i.e. wherever aliases overlap author names, queue the entries for possible merging. 

The question about who should be able to do this is an interesting one I feel might go beyond this specific issue. As of now, due to its small size and devoted community, spam, vandalism and ill intentioned manipulation of data are not an issue on BookWyrm, but if Mastodon (or Goodreads, for that matter) is any reference, that will change once it gains more traction.

**Comment by hughrun** [comment]
@kopischke yep, the existing name-or-alias query is pretty good. The problem is that names are not unique! So without a truly unique identifier you really need a human to eyeball each potential match at some point. My comment you quote was probably a bit unclear: what I mean is that there is no way to _automatically in a guaranteed-to-be-correct_ way to merge records. 

We definitely need another piece of functionality to _manually_ merge them on the basis of some auto-generated helpful hints regarding potential matches - but I don't think the place for that is where I was working this time (the "edit book" workflow).

I realise this may look like a backwards way to come at the problem but I figured it's easier to clean up the old mess if you've at least partially stemmed the flow of new mess coming in.

**Comment by kopischke** [comment]
> The problem is that names are not unique! So without a truly unique identifier you really need a human to eyeball each potential match at some point.

@hughrun totally agree, we can’t and shouldn’t do that automatically. My point was that once we have a system in place to attach an ISNI ID to a BookWyrm Author entity, which this PR provides, we’ll have a starting point for manual merging. By identifying Authors not having an ISNI ID attached, but having an alias matching one or more of those who do, that gives us a pretty good base for identifying merge candidates. And yes, that queue should absolutely be reviewed manually – there’s simply too much context (and possible research for context) to mind for anything else.

> I realise this may look like a backwards way to come at the problem but I figured it's easier to clean up the old mess if you've at least partially stemmed the flow of new mess coming in.

Again, couldn’t agree more, and I think it actually is the right way around. I was just trying to get the ball rolling for the step after that, which might admittedly be a bit premature.

**Comment by TomatDividedBy0** [comment]
> > #1581 should reduce the frequency of this problem but doesn't actually provide a way to merge already-existing records.
> 
> Just spitballing here, but one thing we might be able to leverage is the existing alias system, i.e. wherever aliases overlap author names, queue the entries for possible merging.
> 
> > it's probably something we'd want admins to do rather than just anyone
> 
> The question about who should be able to do this is an interesting one I feel might go beyond this specific issue. As of now, due to its small size and devoted community, spam, vandalism and ill intentioned manipulation of data are not an issue on BookWyrm, but if Mastodon (or Goodreads, for that matter) is any reference, that will change once it gains more traction.


If trust is an issue, there's ways to manage that while still getting the benefits of crowdsourcing the archival, some long-term ideas I can spitball:

- A publicly viewable audit log with a list of changes made and a function to report/undo changes
- The ability to propose an author-merge which can then be forwarded to a moderator for further consideration.

Although, this does make me wonder? How does the federated model currently handle authors/books, given that you have a common pool of data across multiple instances?

**Comment by jfinkhaeuser** [comment]
I'm not sure if this is considered, but I also see duplicate authors which, based on the works associated with them, are obviously meant to be the same, and with the exact same spelling. It'd be nice to be able to merge them.

But let's not do this automatically! There are at least two distinct authors called "Rick Wayne", for example! These need to be kept separate. It'd be similarly good to have an option to, essentially, say "this book is actually by a different author of the same name" - that's possible by editing the book, removing the author, re-adding the author, and *then* choosing "this is a new author". A bit cumbersome, but at least it already works.

**Comment by mouse-reeve** [comment]
I do plan to add some automatic merging, but it will never happen based on author name -- entries would only be merged if they shared a much more reliable unique identifier like the same ISNI or wikipedia entry. And I agree that automatic merging doesn't fully address the problem that manual merging would.

**Comment by jfinkhaeuser** [comment]
Another related suggestion: there is, for example, Andre Norton for whom a duplicate entry exists on OpenLibrary. This was imported into bookwyrm, so that some works are attributed to "Andre Norton (duplicate)" (I edited those, there weren't too many).

In addition to the aforementioned aliases, and in light of such issues, it might be good to treat authors a bit like the work/edition split. That is, have a canonical author profile to which other profiles can be linked as duplicates.

When editing an author profile, I imagine there are two options, both of which have a use:
- Add duplicate profiles (UI similar to how authors are added to editions); adding even a single duplicate makes the profile a canonical profile.
- Display similar named profiles; selecting one makes the current profile a duplicate of the selected.

Furthermore:
- In the above *as well as* when editing or adding editions, prefer canonical profiles. That should help de-duplicate profiles.
- When visiting *any* of the profiles, list the profiles from related (canonical and their duplicates) authors. It may be worth putting these into different sections, e.g. locally added first, then canonical, then duplicates or some such.
- Still add the ability to manually merge profiles. This helps fix mistakes

The rationale here is that it'd be perfectly possible to keep distinct spellings as distinct profiles, if it's a language thing that keeps producing these things - but if it's just users being inattentive or import sources having issues, you can converge on a saner data set over time.

**Comment by mxamber** [comment]
Any progress here? There are two Annika Brockschmidts and half a dozen Mark Brandis on bookwyrm.social (not to mention garbage imports from OpenLibrary and Inventaire) and importing my collection is currently being a major pain in the arse, because even though all books are attributed correctly, the empty author profiles with 0 books won't go away.

**Comment by skobkin** [comment]
I'm regularly having the same problem with Japanese light novels while importing them to the instance I have account on. After import new books are bound to wrong author and I need to delete and add author again for each book to keep "series" working.
It also could be a good idea to allow selecting authors on import like when adding new books.

**Comment by arkhi** [comment]
> I'm regularly having the same problem with Japanese light novels while importing them to the instance I have account on. After import new books are bound to wrong author and I need to delete and add author again for each book to keep "series" working. It also could be a good idea to allow selecting authors on import like when adding new books.

I have been doing similar work on bookwyrm.social. Every time a book is imported into Bookwyrm, it creates a new author, regardless of an existing author with the same name exists or not.

Would it be possible to hook the existing code about existing author (when adding or editing a book) into the import workflow?

**Comment by joric3** [comment]
Dealing with duplicates is still cumbersome. As a short-term improvement, it would be helpful if the choices to link an author to existing entries (or declare it's a new person) would list how many additional info (aliases, external IDs...) the existing entries each have, or how many titles the existing authors are already linked to. When entering new books, the UI currently doesn't help me deciding which of the existing entries (e.g. four which are all for the same author) is the best curated entry.

A long-term improvement definitely would be to find a way to curate author entries across instances by linking them to ISNI IDs or other identifiers (maybe Wikidata and/or VIAF). Person entries without external IDs should be a "last resort"

**Comment by naught101** [comment]
> this is one of the hard problems of bibliographic data science

Perhaps it would be worth adding the ability for normal users to be able to suggest merges, which could then be reviewed by moderators? This would have a higher chance of being right than anything automated, I think.

Also, MusicBrainz might be a good example to look at for some of how to manage duplicate-sounding authors. It uses a de-duplication system that adds extra metadata to an artist to allow disambiguation. The disambiguation text is usually free-form, but often contains information on artists' genre, dates, or place of origin. So Book Wyrm might have something like (examples from [here](https://www.librarything.com/topic/8029)):

- Robert Ellis (crime/mystery author, USA, 1954-)
- Robert Ellis (mathematics textbooks)
- Robert A. Ellis (theology professor, Oxford UK)

Texts like this are what pop-up in the autocomplete when you're adding a new recording/work on MusicBrainz.

**Comment by JohnLines** [comment]
In the cases where the authors have the same Wikipedia entry (easier to discover and enter, but ambiguous for language), or Wikidata entry, they could be merged, but a manual method would safer. 
 A page for Admins as a result of an Author search which showed all the matching Authors with check boxes to select all the  Authors who are identical would help. Then merge anything from the other entries which is not already in the first entry into that, and offer a choice for any conflicting information. Then update all the records in the Books to point to the unified author.

**Comment by JohnLines** [comment]
The original author entries would have to remain, as they may be linked to from another instance, but should return a return a redirect to the unified entry.

**Comment by Magess** [comment]
Will adding the same ISNI to multiple records for the same author automatically merge them? Or at least prompt someone to accept a merge?

I just saw 4 results for Lilith Saintcrow. One had an isni. All the records were for her. I added the isni to the records that didn't have in the hopes of triggering a merge.

**Comment by joric3** [comment]
> Will adding the same ISNI to multiple records for the same author automatically merge them? Or at least prompt someone to accept a merge?

Unfortunately, there's no automatic merge. Contacting your server admin is your best bet.

In my experience importing a book from another instance/catalog will create a new author entry, even if an author entry of the same name already exists on your home server. There may be special cases where no dupe is created, but that's just a guess. The admin on my home server has written a small script to assist with merging, but it's not automatic and still needs manual approval I think.

**Comment by lo48576** [comment]
If the automatic merge feature will be added in future, I strongly hope it to be opt-in (disabled by default).

My native language is non-latin language (Japanese), but the common databases such as ISNI and Open Library often provide book titles and author names primarily in English.
(Other than names, Wikipedia article URLs would also be non-native language of the author.)
Auto-merge could make author names in such "exotic" languages to be rewritten in Latin alphabets, and it would hugely decrease the usability of the BookWyrm instance for the native speakers.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/bookwyrm-social/bookwyrm/issues/1119#issuecomment-2599690207

@lo48576, alternatively, provide duplicate entries for each data type per language. I expect that'll be necessary regardless.

**Comment by joric3** [comment]
Good point about non-Latin scripts!

As a first step to having fewer duplicates, maybe we could get a dialogue when importing a book from somewhere else, similar to the questions asked when creating a new book:

Is this a new author/ the same author as [list of existing entries with similar names]?
Is this a new work/ edition of an existing work?

That's close to existing features. If the goal is to stick to user-curated data on each instance (even though linked open bibliographic data e.g. via worldcat would be great IMO) then a "suggest a merge" feature for authors and books would be useful.

**Comment by lo48576** [comment]
Just an idea: providing a checkbox or radio buttons along with "This is the same author as {{ some entries }}" option would be best if possible, to let the user choose "prefer metadata from the known data source" or "prefer metadata provided by the instance user".

Even a user would like to use some non-latin names, remote data such as ISNI number and URLs will still be useful. Merge suggestion would be possible while respecting the user-provided data.

**Comment by Magess** [comment]
This is tangential probably BUT the ISNI database does store alternative
scripts and spellings for many authors. And their database is queryable
via an API for a low volume of queries. If a user has gone through the
trouble of adding an ISNI to a record, perhaps the ISNI database could
be queried and all those additional names and spellings added to the
Bookwyrm data? I'm not sure how you'd manage the preferred one for your
server. But I wanted to be sure people knew these things existed in the
ISNI records. 

-Lora

On January 18, 2025, Marisa DeMeglio ***@***.***> wrote:
> Just an idea: providing a checkbox or radio buttons along with "This
> is the same author as {{ some entries }}" option would be best if
> possible, to let the user choose "prefer metadata from the known data
> source" or "prefer metadata provided by the instance user".
>
> Even a user would like to use some non-latin names, remote data such
> as ISNI number and URLs will still be useful. Merge suggestion would
> be possible while respecting the user-provided data.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/bookwyrm-social/bookwyrm/issues/1119#issuecomment-
2599697921>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ACCWTDCUAWI63D67MUAI5YL2LJCR5AVCNFSM6AAAAABJKGVVW2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJZGY4TOOJSGE>.
You are receiving this because you commented.Message ID: <bookwyrm-
***@***.***>

**Comment by lo48576** [comment]
> This is tangential probably BUT the ISNI database does store alternative
> scripts and spellings for many authors. And their database is queryable
> via an API for a low volume of queries. If a user has gone through the
> trouble of adding an ISNI to a record, perhaps the ISNI database could
> be queried and all those additional names and spellings added to the
> Bookwyrm data? I'm not sure how you'd manage the preferred one for your
> server. But I wanted to be sure people knew these things existed in the
> ISNI records. 

Yeah I know that the database has multiple spellings, and that's the very reason I get unwanted override of the author names when I chose to reuse the known remote author data on the current implementation.

For example, when I register a new book of the new author `芝村裕吏`, then I need to use that string for the author, but BookWyrm retrieves [ISNI 0000000381385477 Shibamura, Yuri](https://isni.oclc.org/cbs/DB=1.2//CMD?ACT=SRCH&IKT=8006&TRM=ISN%3A0000000381385477&TERMS_OF_USE_AGREED=Y&terms_of_use_agree=send&COOKIE=U50,KENDUSER,I28,B0028++++++,SY,NISNI,D1.2,E65cc4a4e-d,A,H1,,3-28,,30-41,,43-59,,65-70,,74-75,R219.98.52.175,FY), and use the first value `Shibamura, Yuri` as the primary `Name` field!
Of course I can edit the author later and type the intended `芝村裕吏` into `Name` field (again), but I'd say that's not the ideal behavior of merging (or usage of the external data) especially for non-latin languages.

In this case `Aliases` field of the author on the BookWyrm is not so helpful for visit-only users because the authors of books are shown by the spelling at the `Name` field.
The idea in my comment above is, providing the option to respect user-provided string as `Name`, instead of using the first record of ISNI as `Name`.

Currently the option will affect "Confirm Book Info" on new book creation, but the same problem/solution will apply to the merging of duplicate authors.

**Comment by Magess** [comment]
That makes sense. 👍

-Lora

On January 18, 2025, Marisa DeMeglio ***@***.***> wrote:
> > This is tangential probably BUT the ISNI database does store
> > alternative
> >  scripts and spellings for many authors. And their database is
> > queryable
> >  via an API for a low volume of queries. If a user has gone through
> > the
> >  trouble of adding an ISNI to a record, perhaps the ISNI database
> > could
> >  be queried and all those additional names and spellings added to
> > the
> >  Bookwyrm data? I'm not sure how you'd manage the preferred one for
> > your
> >  server. But I wanted to be sure people knew these things existed in
> > the
> >  ISNI records.
> >
>
> Yeah I know that the database has multiple spellings, and that's the
> very reason I get unwanted override of the author names when I chose
> to reuse the known remote author data on the current implementation.
>
> For example, when I register a new book of the new author 芝村裕吏, then I
> need to use that string for the author, but BookWyrm retrieves ISNI
> 0000000381385477 Shibamura, Yuri
> <https://isni.oclc.org/cbs/DB=1.2//CMD?ACT=SRCH&IKT=8006&TRM=ISN%3A0000000381385477&TERMS_OF_USE_AGREED=Y&terms_of_use_agree=send&COOKIE=U50,KENDUSER,I28,B0028++++++,SY,NISNI,D1.2,E65cc4a4e-
> d,A,H1,,3-28,,30-41,,43-59,,65-70,,74-75,R219.98.52.175,FY>, and use
> the first value Shibamura, Yuri as the primary Name field!
>  Of course I can edit the author later and type the intended 芝村裕吏 into
> Name field (again), but I'd say that's not the ideal behavior of
> merging (or usage of the external data) especially for non-latin
> languages.
>
> In this case Aliases field of the author on the BookWyrm is not so
> helpful for visit-only users because the authors of books are shown by
> the spelling at the Name field.
>  The idea in my comment above is, providing the option to respect
> user-provided string as Name, instead of using the first record of
> ISNI as Name.
>
> Currently the option will affect "Confirm Book Info" on new book
> creation, but the same problem/solution will apply to the merging of
> duplicate authors.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/bookwyrm-social/bookwyrm/issues/1119#issuecomment-
2599729216>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ACCWTDDQZJIK4XWWR4FBUR32LJNLTAVCNFSM6AAAAABJKGVVW2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJZG4ZDSMRRGY>.
You are receiving this because you commented.Message ID: <bookwyrm-
***@***.***>

**Comment by hughrun** [comment]
> > This is tangential probably BUT the ISNI database does store alternative
> > scripts and spellings for many authors. And their database is queryable
> > via an API for a low volume of queries. If a user has gone through the
> > trouble of adding an ISNI to a record, perhaps the ISNI database could
> > be queried and all those additional names and spellings added to the
> > Bookwyrm data? I'm not sure how you'd manage the preferred one for your
> > server. But I wanted to be sure people knew these things existed in the
> > ISNI records.
> 
> Yeah I know that the database has multiple spellings, and that's the very reason I get unwanted override of the author names when I chose to reuse the known remote author data on the current implementation.
> 
> For example, when I register a new book of the new author `芝村裕吏`, then I'd like to use that string for the author, but BookWyrm retrieves [ISNI 0000000381385477 Shibamura, Yuri](https://isni.oclc.org/cbs/DB=1.2//CMD?ACT=SRCH&IKT=8006&TRM=ISN%3A0000000381385477&TERMS_OF_USE_AGREED=Y&terms_of_use_agree=send&COOKIE=U50,KENDUSER,I28,B0028++++++,SY,NISNI,D1.2,E65cc4a4e-d,A,H1,,3-28,,30-41,,43-59,,65-70,,74-75,R219.98.52.175,FY), and uses the first value `Shibamura, Yuri` as the primary `Name` field! Of course I can edit the author later and type the intended `芝村裕吏` into `Name` field (again), but I'd say that's not the ideal behavior of merging (or usage of the external data) especially for non-latin languages.
> 
> In this case `Aliases` field of the author on the BookWyrm is not so helpful for visit-only users because the authors of books are shown by the spelling at the `Name` field. The idea in my comment above is, providing the option to respect user-provided string as `Name`, instead of using the first record of ISNI as `Name`.
> 
> Currently the option will affect "Confirm Book Info" on new book creation, but the same problem/solution will apply to the merging of duplicate authors. User should be able to choose one of existing `Name` fields (or also other fields) for the same author on merging.

I *think* I agree we need to align importing with creating of books (i.e. try to reduce duplication of author records),but everyone needs to understand that will create significantly more friction - imagine having to manually confirm every book you import from Goodreads, for example.

We have to choose *something* to display to each user. At the moment we are indeed preferring ISNI for books created within BookWyrm, which whilst "international" is really an English-language biased system, so I hear what @lo48576 is saying. On the other hand, 芝村裕吏 is not legible to me. The ideal scenario is that the author name is presented in the most appropriate way for each user's interface language, and if we don't have that, it falls back to the author's language or however it was originally entered. But our data is not currently arranged in a way that supports that. Ultimately this will be a trade-off between consistency and ease-of-use.

**Comment by beedell-roke** [comment]
> if we choose one name to use, there will always be someone for whom that's a bad choice.

@hughrun, that's patently untrue.

This whole thread seems to be a discussion amongst those unfamiliar with the relative ease by which a data structure can be localised. This is merely and solely a data storage (organisation and extensibility) problem, for its counterparts have been solved:

1. Designing the GUI around it merely entails a combo-box for each localised value that automatically defaults to the browser locale. [^1] [^2]

3. Federating the values is already possible via ActivityPub. [^3] [^4]

Current information, as indeterminately localised as it is, can automatically be assigned `und`, and then manually tagged when next modified. [^5]

I'm unfamiliar with the codebase, but if you consider that the current values are stored somewhat like the undermentioned demonstrates:

~~~YAML
Title: War of the Worlds
~~~

…the simple solution is to generate subkeys when a user enters localised information:

~~~YAML
Title: War of the Worlds
  eng-USA: War of the Worlds
  jpn-JPN: 宇宙戦争
~~~

[^1]: Mastodon (no date a) Add a way for the user to select which languages they understand · issue #29989 · Mastodon/Mastodon, GitHub. Available at: https://github.com/mastodon/mastodon/issues/29989#issue-2251599441:~:text=Insights-,Add%20a%20way%20for%20the%20user%20to%20select%20which%20languages%20they%20understand,-%2329989 (Accessed: 19 January 2025). 

[^2]: Mastodon (no date b) Default to languages indicated by browser · issue #27047 · Mastodon/Mastodon, GitHub. Available at: https://github.com/mastodon/mastodon/issues/27047#issue-1907973550:~:text=Default%20to%20languages%20indicated%20by%20browser (Accessed: 19 January 2025). 

[^3]: Mastodon (no date c) Expose other users’ default/primary language · ISSUE #23067 · Mastodon/Mastodon, GitHub. Available at: https://github.com/mastodon/mastodon/issues/23067#issue-1531127419:~:text=Currently%2C%20posts%20have%20a%20language,for%20profiles%20as%20a%20whole. (Accessed: 19 January 2025). 

[^4]: Mastodon (no date d) Per-toot language · issue #17073 · Mastodon/Mastodon, GitHub. Available at: https://github.com/mastodon/mastodon/issues/17073#issue-1066219645:~:text=Insights-,Per%2Dtoot%20language,-%2317073 (Accessed: 19 January 2025). 

[^5]: LemmyNet (no date e) Remove undetermined from language selection · issue #1166 · LemmyNet/Lemmy-ui, GitHub. Available at: https://github.com/LemmyNet/lemmy-ui/issues/1166#issuecomment-2600835381:~:text=many%20instances%20exist%20in%20which,at%20best%20for%20such%20content. (Accessed: 19 January 2025).

**Comment by hughrun** [comment]
@RokeJulianLockhart I'm confused about what you're saying is "patently untrue", since we're both proposing the exact same solution.

**Comment by beedell-roke** [comment]
> [`#issuecomment-2600985521`](https://github.com/bookwyrm-social/bookwyrm/issues/1119#issuecomment-2600985521)

@hughrun, no worries then. Good to hear. Thanks.

**Comment by luanmdlima** [comment]
Is there anyone currently working on this item? How do I check it, besides coming here and asking?

**Comment by beedell-roke** [comment]
> https://github.com/bookwyrm-social/bookwyrm/issues/1119#issuecomment-3012762427

@luanmdlima, subscribe to the issue. You can see in https://github.com/bookwyrm-social/bookwyrm/issues/1119#event-1144463626 and https://github.com/bookwyrm-social/bookwyrm/issues/1119#event-2679351221 that work has been linked to this issue. When work occurs, you'll see it linked as an event. In future, ask such questions at [`orgs/community/discussions/new`](https://github.com/orgs/community/discussions/new/choose), unless you've reason to believe that it'll differ for a specific project (like if it has no PRs).

**Comment by MMachado05** [comment]
In case it hasn't been considered, could each book/author entry potentially have some button off somewhere that says "Suggest Duplicate"? That is, on click, the user can provide the link to some other object on an instance (e.g. a duplicate book that would be better stored as a separate edition) and then the instance moderators would have some admin view of the duplicate suggestions, wherein they would be able to "accept" the duplicates, thus merging the two into one object (and, in the case of books, creating new editions, or deduping where common values are found, e.g. ISBNs), or reject them.

It kind of offloads a lot of responsibility onto admins, but I figure it's one more route to consider.

**Comment by beedell-roke** [comment]
@MMachado05, I believe that that's what [`internetarchive/openlibrary/issues/10748#issuecomment-2887149361`](https://github.com/internetarchive/openlibrary/issues/10748#issuecomment-2887149361), and similar, request elsewhere. Considering that *that* source is where Bookwyrm polls some of its data from, this might be indirectly improved when the data sources gain their own ways to report duplicates, too.

**Comment by MMachado05** [comment]
That definitely seems promising, but it does seem to have stagnated; recent conversations are dated back to 2024. Would providing a "temporary fix" as I described in my original comment be a path worth pursuing?

**Comment by bitboxer** [comment]
Yes, having the source handle duplicates would be nice. BUT you could import the same book from different sources in Bookwyrm. I doubt that this can be solved without proper duplicate handling inside this project.

**Comment by gary-host-laptop** [comment]
Is this or the merging of books being worked on? If so, what's the PR to track this?

**Comment by beedell-roke** [comment]
@gary-host-laptop, if one existed, it would be linked.

**Comment by hughrun** [comment]
I want to clear up Series and contribution-types first (see https://github.com/bookwyrm-social/bookwyrm/issues/3681). Series has a PR for review currently. The reason for that is my expected changes to contributors will change the Author class.

I agree this is the most pressing issue after that and of course if anyone else wants to start work on it earlier, feel free.

It's also the sort of thing where if we get a fix – even a "quick" one – wrong, it could make things worse because of how data flows between instances. I'd say @MMachado05's suggestion here: https://github.com/bookwyrm-social/bookwyrm/issues/1119#issuecomment-3530854986 is the way to go initially, and we can build up functionality from there.
